internal/cache/metadata/type_cache.go

```diff
@@ -100,20 +113,20 @@ type typeCache struct {
 	entries *lru.Cache
 }
 
-// NewTypeCache creates an LRU-policy-based cache with given max-entries and TTL.
+// NewTypeCache creates an LRU-policy-based cache with given max-size and TTL.
 // Any entry whose TTL has expired, is removed from the cache on next access (Get).
-// When insertion of next entry would cause #entries of cache > maxEntries,
+// When insertion of next entry would cause size of cache > sizeInMB,
 // older entries are evicted according to the LRU-policy.
-// If either of TTL or maxEntries is zero, nothing is ever cached.
-func NewTypeCache(maxEntries int, ttl time.Duration) TypeCache {
-	if ttl > 0 && maxEntries != 0 {
-		var lruMaxEntries uint64 = math.MaxUint64 // default for when maxEntries = -1
-		if maxEntries > 0 {
-			lruMaxEntries = uint64(maxEntries)
+// If either of TTL or sizeInMB is zero, nothing is ever cached.
```

Comment:
I would prefer using sizeMB as variable name because the flag is max-size-mb and max-size-in-mb.

Reply:
done